esmvaltool/diag_scripts/shared/__init__.py

```diff
@@ -27,6 +32,19 @@
     'Variables',
     'Datasets',
     'get_cfg',
+    # IO
```

Comment:
I think it would be better to make these functions available in their own modules, i.e. rename _io.py to io.py and just add the 'io' module here. The same for the iris_helpers functions/module. Having a very large number of functions here is confusing because it becomes impossible for users to find things. That's why e.g. all plotting related functions can be found in the 'plot' module.